apps/server/src/modules/authorization/feathers-auth.provider.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { BaseEntity, EntityId } from '@shared/domain';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { NewsTargetModelValue } from '../news/entity';
+import { FeathersServiceProvider } from '../feathers/feathers-service.provider';
+
+interface User {
+	_id: ObjectId;
+	schoolId: ObjectId;
+	permissions: string[];
+}
+
+@Injectable()
+export class FeathersAuthProvider {
+	constructor(private feathersServiceProvider: FeathersServiceProvider) {}
+
+	async getUserSchoolPermissions(userId: EntityId, schoolId: EntityId): Promise<string[]> | never {
+		const user = await this.getUser(userId);
+		// test user is school member
+		const sameSchool = user.schoolId.toString() === schoolId;
+		if (sameSchool && Array.isArray(user.permissions)) {
+			return user.permissions;
+		}
+		return [];
+	}
+
+	async getUserTargetPermissions(
+		userId: EntityId,
+		targetModel: NewsTargetModelValue,
+		targetId: EntityId
+	): Promise<string[]> {
+		const service = this.feathersServiceProvider.getService(`${targetModel}/:scopeId/userPermissions/`);
```

Comment:
I think you should really really rewrite the new logic and not implement some existing stuff. If you are ready with rewrite we delete the feather services and move the clients to new route.

Reply:
right, might not work everywhere, as it is important not to have too many duplications in code for us too. its just a helper to easily be able to iterate the refactoring.